c++/src/sargs/SargsApplier.cc

```diff
@@ -25,7 +25,8 @@ namespace orc {
   uint64_t SargsApplier::findColumn(const Type& type,
                                     const std::string& colName) {
     for (uint64_t i = 0; i != type.getSubtypeCount(); ++i) {
-      if (type.getFieldName(i) == colName) {
+      // Only STRUCT type has field names
+      if (type.getKind() == STRUCT && type.getFieldName(i) == colName) {
```

Comment:
The directly previous code seems to be ORC-40 at 1.7.0.
Does 1.6.x has the same issue, @stiga-huang ?

Reply:
No, predicate push down in the C++ reader is supported since 1.7.0. ORC-1.6.x doesn't have these codes.

Reply:
Great. Thank you for the confirmation.